### PR TITLE
[1.0-beta2] DeepMind: Add ID to ACCEPT_BLOCK_V2

### LINKED ISSUE #261: Deep-mind ACCEPT_BLOCK_V2 changes
- Add explicit `id` for easier processing

### PR DESCRIPTION
**Pull request by heifner**
### `ACCEPT_BLOCK_V2`

`ACCEPTED_BLOCK_V2 ${id} ${num} ${lib} ${blk} ${fd}`
`${id}` => block id
`${num}` => block number
`${lib}` => Last irreversible block number
`${blk}` => `signed_block` in fc packed hex format
`${fd}` => `finality_data_t` in fc packed hex format

#### `finality_data_t`
```
struct finality_data_t {
   uint32_t     major_version{light_header_protocol_version_major};
   uint32_t     minor_version{light_header_protocol_version_minor};
   uint32_t     active_finalizer_policy_generation{0};
   uint32_t     final_on_strong_qc_block_num{0};
   digest_type  action_mroot{};
   digest_type  base_digest{};
   std::optional<finalizer_policy> proposed_finalizer_policy; // finalizer policy, if proposed in the block
};
```
```
   struct finalizer_policy {
      uint32_t                         generation = 0; ///< sequentially incrementing version number
      uint64_t                         threshold = 0;  ///< vote weight threshold to finalize blocks
      std::vector<finalizer_authority> finalizers;     ///< Instant Finality voter set
   }
```
```
   struct finalizer_authority {
      std::string  description;
      uint64_t     weight = 0;
      fc::crypto::blslib::bls_public_key  public_key;
   }
```

`bls_public_key` is affine non-montgomery base64url with prefix `PUB_BLS_`

Resolves #261

### CODE REVIEW

**Comment by heifner** [comment]
@fschoell

File: libraries/chain/controller.cpp [hunk 1]
```diff
@@ -3282,19 +3282,20 @@ struct controller_impl {
          }
 
          if (auto* dm_logger = get_deep_mind_logger(false)) {
-            auto fd = head_finality_data();
             apply<void>(chain_head,
```
Comment (linh2931) [inline libraries/chain/controller.cpp:3285]: Can you add a comment this is for transition block?

File: libraries/chain/deep_mind.cpp [hunk 2]
```diff
@@ -83,12 +83,14 @@ namespace eosio::chain {
       );
    }
 
-   void deep_mind_handler::on_accepted_block_v2(block_num_type lib, const signed_block_ptr& b, const finality_data_t& fd)
+   void deep_mind_handler::on_accepted_block_v2(block_num_type lib, const signed_block_ptr& b, const block_id_type& id,
```
Comment (linh2931) [inline libraries/chain/deep_mind.cpp:86]: Not a big deal. Maybe it is nicer the order of parameters matches the order of DM fields.

**Review by linh2931 — APPROVED** [review]

**Review by greg7mdp — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: INTERNALS
summary: Add explicit ID to ACCEPT_BLOCK_V2 to improve DeepMind support.
Note:end

**Pull request merged**